src/CoreServiceProvider.php

```diff
@@ -89,6 +93,7 @@ public function boot()
         Event::listen(PlatformSyncing::class, fn () => BlockProcessor::synching());
         Event::listen(PlatformSynced::class, fn () => BlockProcessor::synchingDone());
         Event::listen(PlatformSyncError::class, fn () => BlockProcessor::synchingDone());
+        Event::listen(TransactionCreated::class, TransactionCheckerListener::class);
```

Comment:
I think it would be better to schedule this cron so that it will run asynchronously rather than listening to events

Reply:
I agree, I think a cron job running every few mins (configurable up to where the data isn't available on-chain any more) would be fine.  We do this on the old platform.  Every 5 mins check which transactions are sill in broadcast state and see if there are any updates on-chain for them, if there are then fetch the data and update the DB, send events etc.  If not then they maybe they get moved to abandoned?

Reply:
Discussed on slack.